modules/qna/src/views/full/Components/TextareaList.tsx

```diff
@@ -0,0 +1,109 @@
+import { Button } from '@blueprintjs/core'
+// @ts-ignore
+import BotpressContentPicker from 'botpress/content-picker'
+// @ts-ignore
+import BotpressContentTypePicker from 'botpress/content-type-picker'
+import { lang, Textarea } from 'botpress/shared'
+import _uniqueId from 'lodash/uniqueId'
+import React, { FC, Fragment, useRef, useState } from 'react'
+
+import style from '../style.scss'
+
+interface Props {
+  updateItems: (items: any) => void
+  items: string[]
+  placeholder: (index: number) => void
+  addItemLabel: string
+  label: string
+  keyPrefix: string
+  showPicker?: boolean
+  initialFocus?: string
+  canAddContent?: boolean
+}
+
+const TextareaList: FC<Props> = props => {
+  const [showPicker, setShowPicker] = useState(false)
+  const focusedElement = useRef(props.initialFocus || '')
+  const { updateItems, keyPrefix, canAddContent, addItemLabel, label, items, placeholder } = props
+
+  // Generating unique keys so we don't need to rerender all the list as soon as we add or delete one element
+  const keys = useRef([...Array(items.length)].map(x => _uniqueId(keyPrefix)))
```

Comment:
I know it changes pretty much nothing still I want to make this comment.

Why create a new array and allocate mem only to loop and create a new array ? (moreover .map creates. a new array so why user spread operator ?

following code does the same right ?
`const keys = useRef([items.map(() => _uniqId(keyPrefix))])`